src/utils.rs

```diff
@@ -43,10 +43,16 @@ pub const SHARED_LIBRARY_EXT: &str = "so";
 ///
 /// If the program includes function identifiers, return those. Otherwise return `f` followed by the
 /// identifier number.
-pub fn generate_function_name(function_id: &FunctionId) -> Cow<str> {
+pub fn generate_function_name(
+    function_id: &FunctionId,
+    is_for_contract_executor: bool,
+) -> Cow<str> {
     // Generic functions can omit their type in the debug_name, leading to multiple functions
     // having the same name, we solve this by adding the id number even if the function has a debug_name
-    if let Some(name) = function_id.debug_name.as_deref() {
+
+    if is_for_contract_executor {
+        Cow::Owned(format!("f{}", function_id.id))
+    } else if let Some(name) = function_id.debug_name.as_deref() {
         Cow::Owned(format!("{}(f{})", name, function_id.id))
     } else {
         Cow::Owned(format!("f{}", function_id.id))
```

Comment:
it's not exactly the same, because if `is_for_contract_executor` is true, it will always be that branch, however removing that and just having the if let else will make the debug name have priority even if `is_for_contract_executor` is true

Reply:
Ohh, I see you point. Thanks.